### PR TITLE
Communications: add Discord, expand on Discourse

### PR DESCRIPTION
**Pull request by ncoghlan**
* move the Diversity statement reference up to the top of the
  page (it's getting a bit hidden as the earlier sections grow)
* add info on joining the Discord server
* expand on the usage of Discourse
* mention and link to the Mailman 3 web gateway

### CODE REVIEW

File: communication.rst [hunk 1]
```diff
@@ -6,6 +6,19 @@ Following Python's Development
...
+Standards of behaviour in these communication channels
+------------------------------------------------------
+We try to foster environments of mutual respect, tolerance and encouragement,
+as described in the PSF's `Diversity Statement`_. Abiding by the guidelines
+in this document and asking questions or posting suggestions in the
+appropriate channels are an excellent way to get started on the mutual respect
+part, greatly increasing the chances of receiving tolerance and encouragement
+in return.
+
+.. _Diversity Statement: https://www.python.org/psf/diversity/
```
Comment (ncoghlan) [inline communication.rst:19]: Note: this section isn't new, I just pulled it up from where it was getting buried a bit with the specific sections on the mailings *and* Discourse *and* Discord all coming first and containing quite a bit of text.

**Comment by ncoghlan** [comment]
@Yhg1s Review request is mostly to check that I got the process for assigning Discord server permissions right, but general editorial review is also welcome :)

File: communication.rst [hunk 2]
```diff
@@ -73,24 +89,80 @@ RSS feed readers.
...
+
+For core developers, a long lived multiple use invitation link for this server
+can be found in the private core developer only section of the Discourse forum.
+
+For triagers and documentarians joining the Discord server, a single use invitation
+link should be generated and sent to them directly.
+
+When first joining the server, new users will only have access to the ``#welcome``
+and ``#rules-and-info`` channels. To link their Discord ID with their project
+role, core developers may update their Steering Council `voter record`_ with
```
Comment (hugovk) [inline communication.rst:141]: This repo is private but returns a 404 if you have no access, which looks like a broken link.

I suggest putting a lock icon next to it, as has been done here: https://devguide.python.org/coredev/#gaining-commit-privileges

```suggestion
role, core developers may update their Steering Council 🔒 `voter record`_ with
```

**Review by Mariatta — APPROVED** [review]
Thanks, yeah I think this looks good.

**Pull request merged**